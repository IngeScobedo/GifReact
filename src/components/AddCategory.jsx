/* eslint-disable react/no-unknown-property */
import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = ({ target }) => {
    console.log(target.value)
    setInputValue(target.value)
  }

  const handleSubmit = (event) => {
    event.preventDefault()
    const value = inputValue.trim()
    console.log('value', value)
    if (value.length <= 1) { return }
    onNewCategory(value)
    setInputValue('')
  }

  return (
    <form aria-label="form" onSubmit={handleSubmit} className="w-full flex justify-center">
        <input
            type="text"
            id="search"
            name="search"
            className='border border-black rounded-[28px] text-center text-black my-4 mx-2 w-[80%] px-2 py-3'
            placeholder='Buscar gifs...'
            value={inputValue}
            onChange={ handleInputChange }
            />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
