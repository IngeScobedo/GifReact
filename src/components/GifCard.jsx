import PropTypes from 'prop-types'

export const GifCard = ({ url, title }) => {
  return (
    <div className="w-full h-[145px] rounded-md border border-slate-200 flex flex-col">
        <div className="h-[70%] w-full rounded-t-md">
            <img src={url} alt={title} className='w-full h-full object-cover rounded-t-md' />
        </div>
        <div className='w-full h-[30%] flex flex-col items-center px-2 py-1 text-center justify-center bg-white rounded-b-md'>
            <h2 className='line-clamp-1'>{title}</h2>
        </div>
    </div>
  )
}

GifCard.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}
