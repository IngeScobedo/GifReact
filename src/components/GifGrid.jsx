import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks'
import { GifCard } from './GifCard'
import './GifGrid.css'

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category)

  return (
    <div className="w-full flex justify-center bg-gray-200">
        <section className="w-full max-w-[1100px] flex flex-col gap-5 px-8 py-3">
            <div className='w-full flex justify-center text-center'>
                <h1
                    className='text-indigo-500 font-bold text-4xl first-letter:uppercase' >{category}
                </h1>
            </div>
            <div className='w-full grid grid-cols-3 gap-4'>
                {
                    !isLoading
                      ? gifs.map(gif => (<GifCard url={gif.url} title={gif.title} key={gif.id}/>))
                      : (
                        <div className='w-full h-20 flex justify-center items-center'>
                            <span className='loader'></span>
                        </div>
                        )
                }
            </div>
        </section>

    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
