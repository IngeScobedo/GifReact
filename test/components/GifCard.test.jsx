import { GifCard } from '../../src/components/GifCard'
const { render, screen } = require('@testing-library/react')

describe('Pruebas en <GifCard/>', () => {
  const title = 'Hunter x Hunter'
  const url = 'https://media0.giphy.com/media/nbB1CV1fazlGo/giphy.gif?cid=e21578914ou1xinu6yqyf0zf8ouch0ms7d526axv20lj645s&rid=giphy.gif&ct=g'

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar la imagen con el url y el alt indicado', () => {
    render(<GifCard title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('debe de mostrar el titulo indicado', () => {
    render(<GifCard title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
