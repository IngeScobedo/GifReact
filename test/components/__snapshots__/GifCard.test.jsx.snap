// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="w-full h-[145px] rounded-md border border-slate-200 flex flex-col"
  >
    <div
      class="h-[70%] w-full rounded-t-md"
    >
      <img
        alt="Hunter x Hunter"
        class="w-full h-full object-cover rounded-t-md"
        src="https://media0.giphy.com/media/nbB1CV1fazlGo/giphy.gif?cid=e21578914ou1xinu6yqyf0zf8ouch0ms7d526axv20lj645s&rid=giphy.gif&ct=g"
      />
    </div>
    <div
      class="w-full h-[30%] flex flex-col items-center px-2 py-1 text-center justify-center bg-white rounded-b-md"
    >
      <h2
        class="line-clamp-1"
      >
        Hunter x Hunter
      </h2>
    </div>
  </div>
</div>
`;
